stubs/docutils/docutils/core.pyi

```diff
@@ -1,3 +1,213 @@
-from _typeshed import Incomplete
+from typing import IO, Any, Generic, TypeVar
 
-def __getattr__(name: str) -> Incomplete: ...
+from docutils import SettingsSpec, io, nodes
+from docutils.frontend import OptionParser, Values
+from docutils.io import Input, Output
+from docutils.parsers import Parser
+from docutils.readers import Reader
+from docutils.writers import Writer
+
+__docformat__: str
+
+_S = TypeVar("_S")
+
+class Publisher(Generic[_S]):
+    document: nodes.document | None
+    reader: Reader[_S] | None
+    parser: Parser | None
+    writer: Writer | None
+    source: Input[_S] | None
+    source_class: type[Input[_S]]
+    destination: Output | None
+    destination_class: type[Output]
+    settings: Values | None
+    def __init__(
+        self,
+        reader: Reader[_S] | None = None,
+        parser: Parser | None = None,
+        writer: Writer | None = None,
+        source: Input[_S] | None = None,
+        source_class: type[Input[_S]] = io.FileInput,
```

Comment:
There's some mypy errors about this. It's probably difficult to get this right without overloads.